src/v/raft/consensus.cc

```diff
@@ -2011,4 +2045,37 @@ void consensus::suppress_heartbeats(
     }
 }
 
+voter_priority consensus::next_target_priority() {
+    return voter_priority(
+      std::max<int32_t>(_target_priority * target_priority_update_factor, 1));
+}
+
+/**
+ * We use simple policy where we calculate priority based on the position of the
+ * node in configuration broker vector. We shuffle brokers in raft configuration
+ * so it should give us fairly even distribution of leaders across the nodes.
+ */
+voter_priority consensus::get_node_priority(model::node_id id) const {
+    auto latest_cfg = _configuration_manager.get_latest();
+    auto& brokers = latest_cfg.brokers();
+
+    auto it = std::find_if(
```

Comment:
I think we should add a method to configuration called `find_broker(node_id)` that does this and returns a pointer or an optional

Reply:
there is one already.

Reply:
bu in here I want to do: 

```
std::distance(brokers.cbegin(), it);
```
this is why I need an itearator